src/stores/spaces/SpaceStore.ts

```diff
@@ -162,6 +164,14 @@ export class SpaceStoreClass extends AsyncStoreWithClient<EmptyObject> {
         SettingsStore.monitorSetting("feature_dynamic_room_predecessors", null);
     }
 
+    /**
+     * Returns a promise that resolves when the space store is ready.
+     * This happens after an initial hierarchy of spaces and rooms has been computed.
+     */
+    public get isReady(): Promise<void> {
```

Comment:
This is to prevent the room list store from calling methods on the space store before the space store has computed an initial space hierarchy.